depends/packages/openssl.mk

```diff
@@ -1,9 +1,10 @@
 package=openssl
-$(package)_version=1.0.1
-$(package)_version_suffix=l
+$(package)_version=1.0.2
+$(package)_version_suffix=u
 $(package)_download_path=https://www.openssl.org/source/old/$($(package)_version)
 $(package)_file_name=$(package)-$($(package)_version)$($(package)_version_suffix).tar.gz
-$(package)_sha256_hash=b2cf4d48fe5d49f240c61c9e624193a6f232b5ed0baf010681e725963c40d1d4
+$(package)_sha256_hash=ecd0c6ffb493dd06707d38b14bb4d8c2288bb7033735606569d8f90f89669d16
+$(package)_patches=secure_getenv.patch
 
 define $(package)_set_vars
 $(package)_config_env=AR="$($(package)_ar)" RANLIB="$($(package)_ranlib)" CC="$($(package)_cc)"
```

Comment:
In the following line, variable `$(package)_cc` never defined, using clang by default.

Reply:
By defining `$(package)_cc` by gcc, getting the following error : 
```
gcc: error: x86_64: No such file or directory
gcc: error: unrecognized command line option '-arch'
```

Makefile have the following flags : 
```
CC= gcc
CFLAG= -DOPENSSL_THREADS -D_REENTRANT -pipe -O2 -I/home/ubuntu/build/dogecoin/depends \
        /x86_64-apple-darwin11/include -arch x86_64 -O3 -DL_ENDIAN -Wall -DOPENSSL_IA32_SSE2 \
        -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m \
        -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM \
        -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM
```

Reply:
Hi!

The problem is that because of `make depend`, `config.site` variables get ignored, and there's a little hack in `depends/Makefile` that sets the path using that. This will be a problem with bionic building too; the CI uses the system-installed clang for compiling openssl right now, which is worse because that will introduce further issues with the deterministic build. There's been some work on https://github.com/bitcoin/bitcoin/commit/880660acfa547558f6ef5adff6768de95e53af6e for 0.22 but that may not solve our entire problem.

Cannot solve this by switching to gcc for macOS building either (because that will not cross-compile for Mac) so I'm looking at ways to use the config.site for `make depend`. If you have a fix that does not break determinism of the build, please let us know.

Reply:
I will try to implement the related issue https://github.com/bitcoin/bitcoin/pull/19683 who introduces this commit, try my luck to see if I can get something but looks tricky.

Reply:
I've already done that and am testing right now. You can see my CI run here: https://github.com/patricklodder/dogecoin/runs/3675451209

Reply:
Passing where I was crashing :) On mac logs :
```
*** Because of configuration changes, you MUST do the following before
*** building:

	make depend
make[1]: Entering directory '/home/runner/work/dogecoin/dogecoin/depends/work/build/x86_64-apple-darwin11/openssl/1.0.2-467067eb724'
making depend in crypto...
make[2]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[2]: Entering directory '/home/runner/work/dogecoin/dogecoin/depends/work/build/x86_64-apple-darwin11/openssl/1.0.2-467067eb724/crypto'
making depend in crypto/objects...
make[3]: Entering directory '/home/runner/work/dogecoin/dogecoin/depends/work/build/x86_64-apple-darwin11/openssl/1.0.2-467067eb724/crypto/objects'
../../util/domd ../.. -MD /home/runner/work/dogecoin/dogecoin/depends/x86_64-apple-darwin11/native/bin/clang -target x86_64-apple-darwin11 -mmacosx-version-min=10.8 --sysroot /home/runner/work/dogecoin/dogecoin/depends/SDKs/MacOSX10.11.sdk -mlinker-version=253.9 -- -DOPENSSL_THREADS -D_REENTRANT -pipe -O2 -I/home/runner/work/dogecoin/dogecoin/depends/x86_64-apple-darwin11/include -arch x86_64 -O3 -DL_ENDIAN -Wall -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -DOPENSSL_NO_DEPRECATED -DOPENSSL_NO_CAMELLIA -DOPENSSL_NO_CAPIENG -DOPENSSL_NO_CAST -DOPENSSL_NO_COMP -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_EC_NISTP_64_GCC_128 -DOPENSSL_NO_GMP -DOPENSSL_NO_GOST -DOPENSSL_NO_HEARTBEATS -DOPENSSL_NO_IDEA -DOPENSSL_NO_JPAKE -DOPENSSL_NO_LIBUNBOUND -DOPENSSL_NO_MD2 -DOPENSSL_NO_MDC2 -DOPENSSL_NO_RC4 -DOPENSSL_NO_RC5 -DOPENSSL_NO_RDRAND -DOPENSSL_NO_RFC3779 -DOPENSSL_NO_RSAX -DOPENSSL_NO_SCTP -DOPENSSL_NO_SEED -DOPENSSL_NO_SHA0 -DOPENSSL_NO_SSL_TRACE -DOPENSSL_NO_SSL2 -DOPENSSL_NO_SSL3 -DOPENSSL_NO_STATIC_ENGINE -DOPENSSL_NO_STORE -DOPENSSL_NO_UNIT_TEST -DOPENSSL_NO_WEAK_SSL_CIPHERS -DOPENSSL_NO_WHIRLPOOL --  o_names.c obj_dat.c obj_lib.c obj_err.c obj_xref.c
make[3]: Leaving directory '/home/runner/work/dogecoin/dogecoin/depends/work/build/x86_64-apple-darwin11/openssl/1.0.2-467067eb724/crypto/objects'
making depend in crypto/md4...
make[3]: Entering directory '/home/runner/work/dogecoin/dogecoin/depends/work/build/x86_64-apple-darwin11/openssl/1.0.2-467067eb724/crypto/md4'
```

Reply:
Yes! Though that passed before (it passed differently because it used the wrong clang) so I'm testing all gitian builds before I add the commit here (for both docker and lxc build hosts)